amazon-kinesis-client/src/test/java/software/amazon/kinesis/config/KCLAppConfig.java

```diff
@@ -0,0 +1,197 @@
+package software.amazon.kinesis.config;
+
+import lombok.Value;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.kinesis.utils.RecordValidatorQueue;
+import software.amazon.kinesis.utils.TestRecordProcessorFactory;
+import lombok.Builder;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
+import software.amazon.awssdk.http.Protocol;
+import software.amazon.awssdk.http.SdkHttpConfigurationOption;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClientBuilder;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClientBuilder;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClientBuilder;
+import software.amazon.awssdk.utils.AttributeMap;
+import software.amazon.kinesis.common.ConfigsBuilder;
+import software.amazon.kinesis.common.InitialPositionInStream;
+import software.amazon.kinesis.processor.ShardRecordProcessorFactory;
+import software.amazon.kinesis.retrieval.RetrievalConfig;
+
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.util.Optional;
+
+public interface KCLAppConfig {
+
+    String getStreamName();
+
+    default String getStreamArn() {
+        return null;
+    }
+
+    int getShardCount();
+
+    String getApplicationName();
+
+    String getEndpoint();
+
+    Region getRegion();
+
+
+    /**
+     * "default" profile, should match with profiles listed in "cat ~/.aws/config"
+     */
+    String getProfile();
+
+    InitialPositionInStream getInitialPosition();
+
+    Protocol getConsumerProtocol();
+
+    Protocol getProducerProtocol();
+
+    ProducerConfig getProducerConfig();
+
+    ReshardConfig getReshardConfig();
+
+    default KinesisAsyncClient buildConsumerClient() throws URISyntaxException, IOException {
+        return buildAsyncKinesisClient(getConsumerProtocol());
+    }
+
+    default KinesisAsyncClient buildProducerClient() throws URISyntaxException, IOException {
+        return buildAsyncKinesisClient(getProducerProtocol());
+    }
```

Comment:
We specify two different clients because in many cases, they will have different protocols. The producer will always have a protocol of HTTP1. For KCL 2.x, the consumer can have either HTTP1 or HTTP2 (if it is HTTP2, the two clients cannot be the same).